src/med_bench/mediation.py

```diff
@@ -106,159 +107,37 @@ def mediation_IPW(y, t, m, x, w, z, trim, logit, regularization=True, forest=Fal
-                    rf_xmw_clf = RandomForestClassifier(n_estimators=100,
-                                                        min_samples_leaf=10)
-                if calibration:
-                    p_x_clf = CalibratedClassifierCV(rf_x_clf,
-                                                     method=calib_method)\
-                        .fit(x[train_index, :], t[train_index])
-                    p_wx_clf = CalibratedClassifierCV(rf_xw_clf,
-                                                      method=calib_method)\
-                        .fit(np.hstack((x, w))[train_index, :], t[train_index])
-                    p_xmw_clf = CalibratedClassifierCV(rf_xmw_clf,
-                                                       method=calib_method)\
-                        .fit(np.hstack((x, m, w))[train_index, :],
-                             t[train_index])
-                else:
-                    p_x_clf = rf_x_clf
-                    p_wx_clf = rf_xw_clf
-                    p_xmw_clf = rf_xmw_clf
-                p_x[test_index] = p_x_clf.predict_proba(x[test_index, :])[:, 1]
-                p_wx[test_index] = p_wx_clf.predict_proba(
-                    np.hstack((x, w))[test_index, :])[:, 1]
-                p_xmw[test_index] = p_xmw_clf.predict_proba(
-                    np.hstack((x, m, w))[test_index, :])[:, 1]
-
-            # trimming. Following causal weight code, not sure I understand
-            # why we trim only on p_xm and not on p_x
-            ind = ((p_xmw > trim) & (p_xmw < (1 - trim)))
-            y, t, p_x, p_wx, p_xmw = (y[ind], t[ind], p_x[ind], p_wx[ind],
-                                      p_xmw[ind])
-
-            p_x = np.clip(p_x, clip, 1 - clip)
-            p_xmw = np.clip(p_xmw, clip, 1 - clip)
-            p_wx = np.clip(p_wx, clip, 1 - clip)
-
-            # computation of effects
-            y1m1 = np.sum(y * t / p_x) / np.sum(t / p_x)
-            y1m0 = np.sum(y * t * (1 - p_xmw) / ((1 - p_x) * p_xmw)) /\
-                np.sum(t * (1 - p_xmw) / ((1 - p_x) * p_xmw))
-            y0m0 = np.sum(y * (1 - t) / (1 - p_x)) /\
-                np.sum((1 - t) / (1 - p_x))
-            y0m1 = np.sum(y * (1 - t) * p_xmw / (p_x * (1 - p_xmw))) /\
-                np.sum((1 - t) * p_xmw / (p_x * (1 - p_xmw)))
-            y1m0p = np.sum(y * t / p_xmw * (1 - p_xmw) / (1 - p_wx) * p_wx / p_x)/\
-                np.sum(t / p_xmw * (1 - p_xmw) / (1 - p_wx) * p_wx / p_x)
-            y0m1p = np.sum(y * (1 - t) / (1 - p_xmw) * p_xmw / p_wx * (1 - p_wx) / (1 - p_x)) /\
-                np.sum((1 - t) / (1 - p_xmw) * p_xmw / p_wx * (1 - p_wx) / (1 - p_x))
-
-            return(y1m1 - y0m0,
-                   y1m1 - y0m1,
-                   y1m0 - y0m0,
-                   y1m1 - y1m0p,
-                   y0m1p - y0m0,
-                   len(y) - np.sum(ind))
-
+       # trimming. Following causal weight code, not sure I understand
+        # why we trim only on p_xm and not on p_x
```

Comment:
On likely reason is that p_xm is more likely to reach extreme values, but I agree with you that we should do it for p_x too.

Reply:
Noted